lib/Target/TTNN/TTNNToFlatbuffer.cpp

```diff
@@ -250,13 +245,16 @@ std::shared_ptr<void> ttnnToFlatbuffer(Operation *op) {
   (void)result;
 
   auto debugInfo = ::tt::target::CreateDebugInfoDirect(fbb, mlir, cpp.c_str());
-  auto programOffset = ::tt::target::ttnn::CreateProgramDirect(
-      fbb, program.name, &program.inputs, &program.outputs, &program.ops,
-      debugInfo);
 
-  std::vector<::flatbuffers::Offset<::tt::target::ttnn::Program>> programs = {
-      programOffset,
-  };
+  std::vector<::flatbuffers::Offset<::tt::target::ttnn::Program>> programs;
+  module->walk([&](func::FuncOp func) {
```

Comment:
Btw, what's the benefit of using `func`s over modules? Is it just a different type of "delimiter"?

(this thread is just for my understanding, not blocking for the PR)

Reply:
There's only 1 module which can contain multiple funcs, where each func forms a program in the flatbuffer binary.  So the way it was previously written we were walking over all ops in the module to form a program which is incorrect because `module->walk(Operation* op)` will just pattern match all ops even ones that span functions.

Reply:
Are you saying modules can't be nested?

Reply:
I don't think they can, I _think_ there can be multiple in a single file, but they can't be nested.